kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableSchemaEvolutionSuite.scala

```diff
@@ -1160,33 +1200,68 @@ class DeltaTableSchemaEvolutionSuite extends DeltaTableWriteSuiteBase with Colum
     }
   }
 
-  // TODO: need to remove this once we support schema evolution with clustering columns
-  test("Cannot update schema with clustering columns") {
-    withTempDirAndEngine { (tablePath, engine) =>
-      val table = Table.forPath(engine, tablePath)
-      val initialSchema = new StructType()
-        .add("a", StringType.STRING, true)
-        .add("c", IntegerType.INTEGER, true)
-
-      createEmptyTable(
-        engine,
-        tablePath,
-        initialSchema,
-        clusteringCols = List(new Column("c")),
-        tableProperties = Map(
-          TableConfig.COLUMN_MAPPING_MODE.getKey -> "id",
-          TableConfig.ICEBERG_COMPAT_V2_ENABLED.getKey -> "true"))
-
-      val currentSchema = table.getLatestSnapshot(engine).getSchema
-      val newSchema = new StructType()
-        .add("c", StringType.STRING, true, currentSchema.get("c").getMetadata)
-        .add("a", StringType.STRING, true, currentSchema.get("a").getMetadata)
-
-      assertSchemaEvolutionFails[KernelException](
-        table,
-        engine,
-        newSchema,
-        "Update schema for table with clustering columns")
+  val primitiveSchemaWithClusteringColumn = new StructType()
+    .add("clustering_col", IntegerType.INTEGER, fieldMetadataForColumn(1, "clustering_col"))
```

Comment:
nit - but I feel like the physicalName != the logical name for this test case (to make sure we use the right one when they are not the same)

Reply:
Sure I think that's a good call, I'll fix this